### ISSUE TITLE
Navigation: use curriculum topics for breadcrumbs

### ISSUE THREAD

**Issue by Knorrke** [serlo-org/athene2#531]
>Navigation: Lehrplan vs. Themengliederung 
>
>Aus der Gliederung für einen Lehrplan kann ich auf einen Artikel klicken, der im Inhaltsverzeichnis für Themen aufgehängt ist. Beim Sprung vom Lehrplan in einen Artikel geht die Information verloren in welchem Lehrplan ich war.
>
>So kommt man zwar leicht vom Lehrplan zu einem Artikel, aber zurück geht nichts mehr. Für den nächsten Punkt im Lehrplan muss man wieder durch den ganzen Lehrplan navigieren. 

questions: how to do this with caching? What happens when following links in the article?

**Comment by mik-de** [comment]
In doubt this could be done in JS. Using side navigation would be suitable and non-intrusive to site design (id="navigation-content")

**Comment by Knorrke** [comment]
Hey @mik-de thanks for your input!
With JS you mean setting cookies? I think we would need to pass the correct path information serverside anyway? But it might be tricky to get the relevant data... @arekkas whats your take on this?

Also I'm not sure about the side navigation, did you really mean the left one? Can you sketch what you think this would look like? I think the breadcrumbs/subject-navigation (id="breadcrumbs")  might be better suited for this.

**Comment by mik-de** [comment]
@Knorrke Quick answer to the first point. Cookies are not necessary, because such information could be encoded in the URL. Such information is reachable for JS to my knowledge and could look like in the following example.
https://de.serlo.org/mathe/zahlen-groessen/bruchrechnen-dezimalzahlen/brueche/brueche?curriculum=deutschland/bayern/realschule/klasse-7/zweig-1

**Comment by mik-de** [comment]
@Knorrke to you second point

The "breadcrumbs" are already filled with the logic structure. From didactics view this is an information that shouldn't get lost instead. I've you look at the left navigation they are already a navigation to curriculum, but only very general and not pointing to specific class of a curriculum. From the point a reader has selected a specific class this navigation should be set to his specific branch. The example at the bottom shows how such a branch could look like.

In more advanced features the curriculum branch in left navigation could list all topics listed in page "Zweig 1". This would give the reader an omnipresent feedback about what he is learning.

Additional the reader could be warned if he has left his curriculum.

> Deutschland
>> Bayern
>>> Realschule
>>>> Klasse 7
>>>>> Zweig 1
>>>>>> (Nearby elements for Zweig 1)

**Comment by Knorrke** [comment]
Hey, sorry for the late reply and thanks again for your input to this issue. I'm not so sure if this deep structure really fits the left menu. I think the navigations are already very overloaded. To my understanding the breadcrumb navigation is for the usecase to get back in the tree structure and could probably be implemented in a way, that you can get back to the curriculum topics if you got there that way. So as its used for navigation purposes i think that we don't miss anything, if we replace the current one. To see where the article is placed in a logical way there also is the related links on the right, where the first bullets are induced from the topical structure.

some implementational notice:
- The left navigation is not controlled programmatically, but handled in settings in the CMS. So introducing a programmed structure in this doesn't seem right from a technical point of view either.
- I don't really like the idea of passing the structure that has to be rendered as params to the GET query. This seems like a doorway for malicious links and XSS. Sure we can encode it and probably protect from any attacks, but I don't think that this is the best approach and worth it.

I will discuss this in our sprint planning in july,

**Comment by mik-de** [comment]
You are right the deep branch is to much information and even not necessary. Only school type and class are the dominant items that readers are aware of. This is simple so I've made a visual prototype http://imgur.com/a/NjeAF.

If the curriculum is placed on the left or right navigation bar is a question to the UI-X experts. Also it looks doubled to me (A rudimentary prototype http://imgur.com/a/p2kSj). My request is about a way back to the class in some way but not the back button from the browser or a navigation clicking throu many pages.

Implementation:
Classes could be linked with an id by CMS like school type, though curriculum parameters get much simpler.
Making the right side navigation smart is a matter of cost vs. gain balancing.
Especially for user experience testing a simple JS could be used to get a better idea about the gain. 

I'm looking forward for the results of the discussion.

**Comment by Knorrke** [comment]
Hey @mik-de so we discussed this today and we came to the conclusion that the information you'd want about how the user navigated through the website would be really nice to have. However it's a bigger task as simple solutions fail to the theoretically unbounded number of different navigations/curriculas and we currently are running with a very small team in the development and UX design, so this probably is a task to get to in the future. For now sadly the way for users to get back to the class they came from is by using the browsers back button.

Thanks anyway for the rich input you provided already, I feel like this will help very much when we get to the task :-)
